src/player.cpp

```diff
@@ -2097,11 +2097,30 @@ Item* Player::getCorpse(Creature* lastHitCreature, Creature* mostDamageCreature)
 {
 	Item* corpse = Creature::getCorpse(lastHitCreature, mostDamageCreature);
 	if (corpse && corpse->getContainer()) {
+		std::map<std::string, uint16_t> names;
+		for (const auto& killer : getKillers()) {
+			names[killer->getName()]++;
+		}
+
 		std::ostringstream ss;
+		ss << "You recognize " << getNameDescription() << ". " << (getSex() == PLAYERSEX_FEMALE ? "She" : "He") << " was killed by ";
+		size_t countNames = names.size();
 		if (lastHitCreature) {
```

Comment:
if lastHitCreature is nullptr then we will always get the message "something evil", is that intended? I dont think it is.
This would give false info, if for example player dies by field then I think lastHitCreature is nullptr, so the whole code snippet wont get executed and we will end up with "killed by something evil" even though he could have been attacked by other players/monsters and even mostDamageCreature could be a player.

Reply:
if player is killed by nullprt and others (monsters/players) it should say: **She was killed by something evil and others.**
e.g:

> He was killed by something evil and True Gabz and others.
> She was killed by something evil and others.

Reply:
> if player is killed by nullprt and others (monsters/players) it should say: **She was killed by something evil and others.**
> e.g:
> 
> > He was killed by something evil and True Gabz and others.
> > She was killed by something evil and others.

interesting, I will try to improve the code so that this message can happen in such situations.

Reply:
> if lastHitCreature is nullptr then we will always get the message "something evil", is that intended? I dont think it is.
> This would give false info, if for example player dies by field then I think lastHitCreature is nullptr, so the whole code snippet wont get executed and we will end up with "killed by something evil" even though he could have been attacked by other players/monsters and even mostDamageCreature could be a player.

the very name of the variable says it all, the last hit, this does not mean that it is responsible for your death, it is simply the last creature that hit you, the only way for it to be nullptr is for this creature to be eliminated for some reason, such as: inactivity or its own death, this it means that if a field of fire executes you, the last creature is still mentioned as your killer.

In any case, I am implementing changes for when this situation arises ... the situation in which your killer is eliminated and now if you are executed by something evil.